packages/css-engine/src/core/style-sheet-atomic.ts

```diff
@@ -0,0 +1,58 @@
+import type { Style, StyleProperty } from "../schema";
+import { StyleRule } from "./rules";
+import type { TransformValue } from "./to-value";
+import hash from "@emotion/hash";
+import { StyleSheet, type CssRule } from "./style-sheet";
+
+const defaultMediaRuleId = "__default-media-rule__";
+
+export class StyleSheetAtomic extends StyleSheet {
+  addStyleRule(
+    rule: CssRule,
+    selectorSuffix: string = "",
+    transformValue?: TransformValue
+  ) {
+    const mediaRule = this.addMediaRule(rule.breakpoint || defaultMediaRuleId);
+    const styleRules = [];
+    const classes = [];
+
+    let property: StyleProperty;
+    for (property in rule.style) {
+      const value = rule.style[property];
+
+      // We need to create the new rule to be able to compare the selectorText aka hash.
+      const newStyleRule = new StyleRule(
+        "",
+        { [property]: value } as Style,
+        transformValue,
+        this.#onChangeRule
+      );
+      styleRules.push(newStyleRule);
+      // "c" makes sure hash always starts with a letter.
+      const className = `c${hash(
+        newStyleRule.cssText + selectorSuffix + rule.breakpoint
+      )}`;
+      classes.push(className);
+      newStyleRule.selectorText = `.${className}${selectorSuffix}`;
+
+      const ruleExists = mediaRule.rules.some((styleRule) => {
+        return (
+          "selectorText" in styleRule &&
+          // This property-value combination has already been added.
```

Comment:
Is it necessary to check for selectorText existance? Seems like yiu pass empty string by default

Reply:
I set it after that, its not beautiful, I am creating a rule to get cssText and then create hash, alternatively I have to move hash creation to the style rule

Reply:
actually I can just pass a function to a constructor so regular one can return a different one than atomic, so the logic is more clear

Reply:
rewrote